cmd/decode.go

```diff
@@ -95,6 +104,60 @@ func validateAndRun() error {
 	return run(options.metaOnly, outMediaType, in, os.Stdout)
 }
 
+// Original requirement from etcd WAL log analysis tool etcd-dump-logs,
+// to serve as a decoder to decode k8s objects from each entry of the etcd WAL log.
+// Read data from os.Stdin, decode object line by line, and print out in os.Stdout
+func runInBatchMode(metaOnly bool, outMediaType string, out io.Writer) error {
+	var err error
+	inputReader := bufio.NewReader(os.Stdin)
+	for {
+		input, err := inputReader.ReadBytes(byte('\n'))
+		if err == io.EOF {
+			fmt.Printf("EOF\n")
+			return nil
+		}
+		if err != nil {
+			fmt.Printf("Read Error: %s\n", err)
+			return err
+		}
+
+		input = stripNewline(input)
+		decodedinput, err := hex.DecodeString(string(input))
+		if err != nil {
+			fmt.Println(err)
+			return err
+		}
+		inMediaType, decodedinput, err := encoding.DetectAndExtract(decodedinput)
+		if err != nil {
+			os.Stderr.WriteString(err.Error())
+			os.Stderr.WriteString("\n")
```

Comment:
error is printed to os.Stderr, so etcd-dump-logs could differentiate the error from real decoded data the print them out in separate output column.